### PR TITLE
CVE-2013-4444

### PR DESCRIPTION
**Pull request by bxj3964**
CVE-2013-4444 completed

### CODE REVIEW

File: cves/CVE-2013-4444.yml [hunk 1]
```diff
@@ -4,13 +4,13 @@ nickname_instructions: |
...
   alongside its CVE to make it more easily distinguished from the rest.
-nickname: Unrestricted File Upload Remote Code Execution
+nickname:
 
 CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 434-Unrestricted Upload of File with Dangerous Type
+CWE: 94
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml]: I recommend taking a look at CWE 434 as it seems more appropriate in this situation

File: cves/CVE-2013-4444.yml [hunk 2]
```diff
@@ -224,10 +214,8 @@ mistakes:
...
-    Even after doing the research, it is not very clear what happened that
-    lead to this vulnerability other than not checking for edge cases that
-    might cause vulnerabilities such as outdated versions of languages and
-    frameworks that might not be compatible with software being built.
-
-    In addition, it's interesting that the fix wasn't focused on fixing the issue
-    but instead it was an overall "code cleanup" that included the fix.
+    The vulnerability was created because file uploads were overlooked as a 
+    source of malicious code, and the appropriate protections were not 
+    implemented when the feature was added. While this is not a problem that is
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml:223]: is was in the same sentence. Recommend staying in one tense.

File: cves/CVE-2013-4444.yml [hunk 3]
```diff
@@ -131,12 +121,11 @@ discovered:
     may leave the entries blank except for "answer", BUT please write down
     where you looked in "answer".
   answer: |
-    Pierre Ernst of the VMware Security Engineering, Communications and
-    Response group (vSECR) found this vulnerabilithy and reported it to the
-    Tomcat security team via the Pivotal security team.
-  date: 09-05-2014
-  automated:
-  contest:
+    The vulnerability was found and reported Pierre Ernst who is a third party
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml]: reported by

File: cves/CVE-2013-4444.yml [hunk 4]
```diff
@@ -44,14 +44,10 @@ description_instructions: |
...
-  important rather than critical because the circumstances required for it to
-  occur are limited.
-
-  The implementation allowed unrestricted reading of files. It was not obvious
-  that files were checked for scripting. This may allow remote code execution by
-  attackers using JavaScript files.
+  This vulnerability has to do with how Apache handles file uploading. Before the 
+  fix, users could upload any files they want to the system without checking. 
+  Malicious users can take advantage of this by uploading files containing 
+  arbitrary code to the vulnerable computer, and that code will be run.
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml]: It sounds a little awkward. "and the code will be run", instead try "which will execute"

File: cves/CVE-2013-4444.yml [hunk 5]
```diff
@@ -176,8 +165,11 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml:169]: were there any commits you found between the fix and vcc? I recommend running the following: "git log --before={date when fix occured} --stat"

File: cves/CVE-2013-4444.yml [hunk 6]
```diff
@@ -72,10 +68,8 @@ fixes:
    - commit:
      note:
 vccs:
-  - commit: f1911d8c5f0dc3774fe731ae27d4a9a9b39197c3
-    note: |
-      I found this VCC by checking the blame on the file reading reading code
-      and the original commit that introduced it.
+  - commit: 07fee80270e7006cb68cf1689dd66434d939a1c7
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml:71]: Agree on this, as you can see the new file upload handler was implemented

File: cves/CVE-2013-4444.yml [hunk 7]
```diff
@@ -4,13 +4,13 @@ nickname_instructions: |
   Nickname is optional. Provide a useful, professional, and catchy nickname for
   this vulnerability. Ideally fewer than 30 characters. This will be shown
   alongside its CVE to make it more easily distinguished from the rest.
-nickname: Unrestricted File Upload Remote Code Execution
```
Comment (sk6786) [inline cves/CVE-2013-4444.yml:7]: My upvote on this is 3. I think this is a relevant issue and one easily overlooked as we do not anticipate users to upload malicious files.

File: cves/CVE-2013-4444.yml [hunk 8]
```diff
@@ -109,11 +103,7 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer: |
-    The fix didn't involve unit testing, rather it was a "code cleanup" fix as
-    it required deletion of code. I think that not having unit tests
-    makes sense in this case as this vulnerability is an edge case that needs
-    limited conditions such as an outdated java.io.File in order to happen.
+  answer: Looking through the unit tests, I do not see any for file uploading.
```
Comment (jsj1027) [inline cves/CVE-2013-4444.yml]: I think you should mention that the fix was code cleanup as the person who wrote this before did.
